core/src/impl/Kokkos_ClockTic.hpp

```diff
@@ -90,6 +90,26 @@ KOKKOS_IMPL_HOST_FUNCTION inline uint64_t clock_tic_host() noexcept {
 
   return (uint64_t)cycles;
 
+#elif defined(__ppc__)
+
+  uint64_t result = 0;
+  uint32_t upper, lower, tmp;
+
+  __asm__ volatile(
+    "0: \n"
+    "\tmftbu %0     \n"
+    "\tmftb  %1     \n"
+    "\tmftbu %2     \n"
+    "\tcmpw  %2, %0 \n"
+    "\tbne   0b     \n"
+    : "=r"(upper), "=r"(lower), "=r"(tmp)
+    );
+  result = upper;
+  result = result << 32;
+  result = result | lower;
+
+  return(result);
```

Comment:
Please comment some more on where this code is coming from or how it is working.

Reply:
@masterleinad This paper has it, for example: https://pages.cs.wisc.edu/~legault/miniproj-736.pdf
Also here: https://cmssdt.cern.ch/lxr/source/FWCore/Utilities/interface/HRRealTime.h